whirlpool/appliancesmanager.py

```diff
@@ -18,71 +18,108 @@ def __init__(
+            f"{self._backend_selector.base_url}/api/v1/share-accounts/appliances",
             headers=self._create_headers(),
         ) as r:
             if r.status != 200:
-                LOGGER.error(f"Failed to get appliances: {r.status}")
+                LOGGER.error(f"Failed to get shared appliances: {r.status}")
                 return False
 
-            self._aircons = []
-            self._washer_dryers = []
-            self._ovens = []
+            data = await r.json()
+            locations: List[Dict[str, Any]] = data["sharedAppliances"]
+            for appliances in locations:
+                for appliance in appliances["appliances"]:
+                    self._add_appliance(appliance)
 
-            locations = json.loads(await r.text())[str(account_id)]
-            for appliances in locations.values():
-                for appliance in appliances:
-                    appliance_data = {
-                        "SAID": appliance["SAID"],
-                        "NAME": appliance["APPLIANCE_NAME"],
-                        "DATA_MODEL": appliance["DATA_MODEL_KEY"],
-                        "CATEGORY": appliance["CATEGORY_NAME"],
-                        "MODEL_NUMBER": appliance.get("MODEL_NO"),
-                        "SERIAL_NUMBER": appliance.get("SERIAL"),
-                    }
-                    data_model = appliance["DATA_MODEL_KEY"].lower()
-                    if "airconditioner" in data_model:
-                        self._aircons.append(appliance_data)
-                    elif "dryer" in data_model or "washer" in data_model:
-                        self._washer_dryers.append(appliance_data)
-                    elif (
-                        "cooking_minerva" in data_model
-                        or "cooking_vsi" in data_model
-                        or "cooking_u2" in data_model
-                    ):
-                        self._ovens.append(appliance_data)
-                    else:
-                        LOGGER.warning(
-                            "Unsupported appliance data model %s", data_model
-                        )
-        return True
+            return True
+
+    async def _get_account_id(self):
+        if self._account_id:
+            return self._account_id
+
+        async with self._session.get(
+            f"{self._backend_selector.base_url}/api/v1/getUserDetails",
+            headers=self._create_headers(),
+        ) as r:
+            if r.status != 200:
+                LOGGER.error(f"Failed to get account id: {r.status}")
+                return False
+            data = await r.json()
+            self._account_id = data["accountId"]
+
+        return self._account_id
+
+    async def fetch_appliances(self):
+        account_id = await self._get_account_id()
+        success_owned = await self._get_owned_appliances(account_id)
+        success_shared = await self._get_shared_appliances()
+
+        return success_owned and success_shared
```

Comment:
```suggestion
        return success_owned or success_shared
```
In my case, the shared_appliances API fails with a 404, so we cannot enforce it.

Reply:
That's good to know, didn't realize it would return a 404 if there were no shared appliances. I'll update this then

Reply:
I assume that this is because I am using the "old" client credentials or something, so it is not authorized to use the shared appliances endpoint. But who knows :shrug: